### PR TITLE
Release Zebra v1.0.0-beta.2

### LINKED ISSUE #3024: Tag Zebra v1.0.0-beta.2 release
During Sprint 24, we need to tag the Zebra 1.0.0-beta.2 release for the work completed during [Sprint 23](https://github.com/ZcashFoundation/zebra/milestone/32).

### PR DESCRIPTION
**Pull request by teor2345**
Closes #3024

### Which Crates to Increment

To check if any of the top-level crates need version increments, go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra and use the last modified dates of each crate. Alternatively you can use the github compare tool and check the `main` branch against the last tag ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.15...main)). `git diff --stat <previous-release-tag> origin/main` is also useful to see what's changed.

- [x] Increment the crates that have new commits since the last version update
- [x] Increment any crates that depend on crates that have changed
- [x] Use the `zebrad` crate version in the `zebra-network` user agent string
      (currently the constant `USER_AGENT` in `zebra-network/src/constants.rs`)
- [x] Use the latest git tag in `README.md`

### How to Increment Versions

Zebra follows [semantic versioning](https://semver.org).

Semantic versions look like: `MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]`

#### Pre-Release Crates

Pre-Release versions have a `TAG` like "alpha" or "beta". For example: `1.0.0-alpha.0`

1. Increment the `PRE-RELEASE` version for the crate.

Optionally, if a `MINOR` feature pre-release breaks `MAJOR` API compatibility:

2. Increment the `MAJOR` version, and reset all the other versions to zero

#### Unstable Crates

Unstable versions have a `MAJOR` version of zero. For example: `0.1.0`

1. Follow stable crate versioning, but increment the `MINOR` version for breaking changes

#### Stable Crates

For example: `1.0.0`

Increment the first version component in this list, and reset the other components to zero:
1. MAJOR versions for breaking public API changes and removals
    * check for types from dependencies that appear in the public API
2. MINOR versions for new features
3. PATCH versions for bug fixes
    * includes dependency updates that don't impact the public API

### Version Locations

Once you know which versions you want to increment, you can find them in the:
- [x] zebra* `Cargo.toml`s
- [x] tower-* `Cargo.toml`s
- [x] `zebra-network` protocol user agent: https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/constants.rs
- [x] `README.md`
- [x] `Cargo.lock`: automatically generated by `cargo build`

Merge all these version increments as one commit, by squashing and rebasing the PR onto the main branch.

After you have your changes pushed start a PR with them using this template by adding `&template=release-checklist.md` to the comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.16?expand=1&template=release-checklist.md)).

#### Version Tooling

You can use `fastmod` to interactively find and replace versions.

For example, for `zebra-1.0.0-alpha.12`, we did:
```
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-alpha.12' '1.0.0-alpha.13'
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-alpha.11' '1.0.0-alpha.12'
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-alpha.10' '1.0.0-alpha.11'
fastmod --extensions rs,toml,md --fixed-strings '0.2.9' '0.2.10' tower-batch
fastmod --extensions rs,toml,md --fixed-strings '0.2.8' '0.2.9' tower-fallback
```

### Reviewing Version Bump Pull Requests

Check for missed changes by going to:
`https://github.com/ZcashFoundation/zebra/tree/<commit-hash>/`
Where `<commit-hash>` is the hash of the last commit in the version bump PR.

If any Zebra or Tower crates have commit messages that are **not** a version bump, we have missed an update.

Also check for crates that depend on crates that have changed. They should get a version bump as well.

## Initial Testing

- [x] After any changes, test that the `cargo install` command in works. Use
      e.g. `cargo install --locked --path zebrad`.
      
## README

As we resolve various outstanding known issues and implement new functionality with each release, we should double check the README for any necessary updates.

We should check and update if necessary:

- [x] The "Beta Release" section
- [x] The "Known Issues" section 

to ensure that any items that are resolved in the latest release are no longer listed in the README. 

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog. Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases).

To create the final change log:
- [x] Copy the draft changelog into `CHANGELOG.md`
- [x] Delete any trivial changes
- [x] Combine duplicate changes
- [x] Edit change descriptions so they are consistent, and make sense to non-developers
- [x] Check the category for each change
  - prefer the "Fix" category if you're not sure
- [x] Create a draft PR to update `CHANGELOG.md`

#### Change Categories

From "Keep a Changelog":
* `Added` for new features.
* `Changed` for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

## After merging the version update PR
- [x] Check for any PRs that have been merged since you created the draft PR to update `CHANGELOG.md` and push any updates if necessary
- [x] Mark the PR to update `CHANGELOG.md` as "Ready for Review"
- [x] Once the changelog PR has been approved and merged, update the draft release with the final changelog
- [x] Set the release title to `Zebra ` followed by the version tag, for example: `Zebra 1.0.0-alpha.0` 
- [x] Set the tag name to the version tag, for example: `1.0.0-alpha.0`
- [x] Set the release to target the `main` branch
- [x] Mark the release as 'pre-release' (until we are no longer alpha/beta)

## Final Testing

- [x] After tagging the release, test that the exact `cargo install` command in
      `README.md` works (`--git` behaves a bit differently to `--path`)

If the build fails after tagging:
1. fix the build
2. check if the fixes changed any extra crates, and do the required version increments
3. update `README.md` with a **new** git tag
4. update `CHANGELOG.md` with details about the fix
5. tag a **new** release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,53 @@ All notable changes to Zebra are documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-beta.2](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.2) - 2021-12-03
+
+Zebra's latest beta finishes transaction header and amount validation, and continues implementing proof and note commitment tree validation.
```
Comment (teor2345) [inline CHANGELOG.md]:
```suggestion
Zebra's latest beta continues implementing zero-knowledge proof and note commitment tree validation. In this release, we have finished implementing transaction header and amount validation.

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,54 @@ All notable changes to Zebra are documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-beta.2](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.2) - 2021-12-03
+
+Zebra's latest beta continues implementing zero-knowledge proof and note commitment tree validation. In this release, we have finished implementing transaction header and amount validation.
+
```
Comment (teor2345) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,54 @@ All notable changes to Zebra are documented in this file.
...
+- Shut down channels and tasks on PeerSet Drop (#3078)
+- Re-order Zebra startup, so slow services are launched last (#3091)
+- Fix slow Zebra startup times, to reduce CI failures (#3104)
+- Speed up CI, and split unrelated and conflicting CI jobs (#3077)
+
+## Security
+
+- Stop closing connections on unexpected messages, Credit: Equilibrium (#3120)
+- Security: Stop routing inventory requests by peer address (#3090)
```
Comment (teor2345) [inline CHANGELOG.md]:
```suggestion
- Stop routing inventory requests by peer address (#3090)
```

**Comment by teor2345** [comment]
I used `cargo release` to bump the versions, but it didn't actually work that well. I couldn't get it to skip unmodified crates, so I had to do it manually.

```sh
cargo release --no-publish --no-push --no-tag --no-verify --no-dev-version --prev-tag-name v1.0.0-beta.1 --package tower-batch --execute patch

cargo release --no-publish --no-push --no-tag --no-verify --dev-version-ext beta --prev-tag-name v1.0.0-beta.1 --exclude tower-fallback --exclude tower-batch --exclude zebra-rpc --exclude zebra-client --execute beta
```

**Review by mpguerra — DISMISSED** [review]
LGTM pending any new PRs that merge between now and this PR merging

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,53 @@ All notable changes to Zebra are documented in this file.
...
+## Fixed
+
+- Shut down channels and tasks on PeerSet Drop (#3078)
+- Re-order Zebra startup, so slow services are launched last (#3091)
+- Fix slow Zebra startup times, to reduce CI failures (#3104)
+- Speed up CI, and split unrelated and conflicting CI jobs (#3077)
+
+## Security
+
+- Stop closing connections on unexpected messages, Credit: Equilibrium (#3120)
```
Comment (teor2345) [inline CHANGELOG.md]:
```suggestion
- Stop closing connections on unexpected messages, Credit: Equilibrium (#3120, #3131)
```

File: .github/PULL_REQUEST_TEMPLATE/release-checklist.md [hunk 5]
```diff
@@ -135,7 +135,7 @@ From "Keep a Changelog":
 - [ ] Check for any PRs that have been merged since you created the draft PR to update `CHANGELOG.md` and push any updates if necessary
 - [ ] Mark the PR to update `CHANGELOG.md` as "Ready for Review"
 - [ ] Once the changelog PR has been approved and merged, update the draft release with the final changelog
-- [ ] Set the release title to `Zebra ` followed by the version tag, for example: `Zebra 1.0.0-alpha.0` 
+- [ ] Set the release title to `Zebra ` followed by the version tag, for example: `Zebra 1.0.0-alpha.0`
 - [ ] Set the tag name to the version tag, for example: `1.0.0-alpha.0`
```
Comment (teor2345) [inline .github/PULL_REQUEST_TEMPLATE/release-checklist.md]:
```suggestion
- [ ] Set the tag name to the version tag, for example: `v1.0.0-alpha.0`
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,53 @@ All notable changes to Zebra are documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-beta.2](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.2) - 2021-12-03
+
+Zebra's latest beta continues implementing zero-knowledge proof and note commitment tree validation. In this release, we have finished implementing transaction header and amount validation.
```
Comment (teor2345) [inline CHANGELOG.md]:
```suggestion
Zebra's latest beta continues implementing zero-knowledge proof and note commitment tree validation. In this release, we have finished implementing transaction header, transaction amount, and Zebra-specific NU5 validation. (NU5 mainnet validation is waiting on an `orchard` crate update, and some consensus parameter updates.)
```

**Pull request merged**